### ISSUE TITLE
C# CreateTimer() does not get freed

### ISSUE THREAD

**Issue by nvlled** [godotengine/godot#93619]
### Tested versions

Reproducible in:
- v4.3.beta2.mono.official [b75f0485b]
- v4.2.2.stable.mono.official [15073afe3]
- v4.1.4.stable.mono.official [fe0e8e557]
- v4.0.4.stable.mono.official [fc0b241c9]

Not tested in Godot 3. 

### System information

Godot v4.2.2.stable.mono - Ubuntu 22.04.4 LTS 22.04

### Issue description

In C# script, `SceneTimer` created with `Node.GetTree().CreateTimer()`  does not get freed or garbage collected. The expected behaviour is for it to get freed automatically and not accumulate memory.

I discovered this when I left a godot program running for an hour or two. It started to lag, and the CPU usage gradually increased. Memory usage is not noticeable though.

### Steps to reproduce

1. Call CreateTimer() in _Process()

```csharp
public partial class Main : Godot.Node2D
{
    public override void _Process(double delta)
    {
        GetTree().CreateTimer(1);
    }
}
```

2. Open Debugger > Monitors panel to see the object allocated objects gradually increase over time. After 5 minutes, it has accumulated 20,000 objects. No code is running aside from the CreateTimer(). In contrast, the objects stay around 1000 with the gdscript version.

![image](https://github.com/godotengine/godot/assets/916556/20019efa-fdd3-4776-b5a9-3cceb20e168f)

### Minimal reproduction project (MRP)

[create-timer-bug.zip](https://github.com/user-attachments/files/15982875/create-timer-bug.zip)

**Comment by zaevi** [comment]
You can manually dispose these managed reference:
```csharp
public partial class Main : Godot.Node2D
{
    public override void _Process(double delta)
    {
        using var _ = GetTree().CreateTimer(1);
    }
}
```

**Comment by paulloz** [comment]
There seem to be two things here. For versions <= 4.2 (before #87526 was merged), there is indeed a leak happening. That should be fixed.

However, it won't prevent the graph from looking like that until the garbage collector runs. For objects that are reference-counted in Godot, the managed instance increments the counter to keep the object alive. C# uses garbage collection[^1], objects that aren't used aren't released immediately from memory. So, until the collector runs, said objects will still appear in the monitor.

See what happens when the collection is forced: the objects are actually released, and the count immediately drops.
![image](https://github.com/godotengine/godot/assets/437025/2b42e3c9-937a-49fd-9157-1800c4d0daf5)

In addition, this is the MRP running on v4.3.beta.mono.custom_build [93fc9b885]. Without forcing anything, we can see collection happening at some point.
![image](https://github.com/godotengine/godot/assets/437025/7dfd1159-e66d-4362-b48b-2f5202f18726)

I did not, however, experience an increase in CPU usage (either on 4.2 or 4.3). I let 4.2 ramp to ~50k objects without noticing my CPU getting above 4% usage.

[^1]: https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals

**Comment by nvlled** [comment]
Okay, this is indeed fixed in 4.3. I checked yesterday, but didn't wait long enough for the GC to clean up. I need to wait for a minute or two before it gets cleaned up:

![godot4 3](https://github.com/godotengine/godot/assets/916556/242a5c92-44b2-49ee-b839-1da299d435b6)

I can't reproduce the higher CPU usage now, strangely enough. I'm using an older hardware so it tends to be more noticeable for me. But will close the issue anyway since it's already fixed.

**Comment by nvlled** [comment]
> You can manually dispose these managed reference:
> 
> ```cs
> public partial class Main : Godot.Node2D
> {
>     public override void _Process(double delta)
>     {
>         using var _ = GetTree().CreateTimer(1);
>     }
> }
> ```

No effect for me
